<a href="https://colab.research.google.com/github/OllyMuh/Probability-Statistics/blob/main/Probability%20Theory%20%2B%20Combinatorics.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ДЗ Теория вероятности, комбинаторика 06.09.2022

In [ ]:
from math import factorial

In [ ]:
# Число сочетаний из n элементов по k элементов в каждом (в сочетаниях порядок не важен):
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [ ]:
# Определим число размещений из n элементов по k элементов в каждом. 
#При размещениях порядок важен, поэтому вариантов размещения может быть больше, чем сочетаний при заданных k и n.
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

In [ ]:
#Число перестановок из n элементов — при перестановках важен порядок, но отличие от размещений в том, что применяются все имеющиеся n элементов:

def permutations(n):
    return int(factorial(n))

# Задача 1

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

Число сочетаний из n элементов по k элементов в каждом (в сочетаниях порядок не важен):

In [ ]:
combinations(10, 4)

210

Ответ: 210 способов

# Задача 2

Из колоды в 52 карты вынимают случайным образом 4 карты. Найдите число исходов, соответствующих тому, что был вытянут хотя бы один туз.

1. Один туз из четырех тузов

In [ ]:
combinations(4, 1)

4

2. Три карты (оставшиеся) из остальных 52 карт

In [ ]:
combinations(52, 3)

22100

3. В этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбраны четыре, одна из которых — туз:

In [ ]:
combinations(52, 3) * combinations(4, 1)

88400

# Задача 3

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

Всего пар соседей - 6 * Каждая пара человек может сесть двумя способами * Все остальные 5 любыми, то есть 5! способами.
/
Рассадить всех можно 7! способами

In [ ]:
6*2*factorial(5) / factorial(7)

0.2857142857142857

# Задача 4

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает:
Все.
Два.

Формула определения вероятности p = m / n

n - число всех равновозможных исходов, студент выбрал любую тройку вопросов, будем считать, что билеты будут разными, если они отличаются вопросами, а порядок вопросов в билете не учитывается (3 из 60).

In [ ]:
n = combinations(60, 3)
n

34220

1. Студент знает все 3 вопроса

m  - число элементарных исходов, благоприятствующих событию (3 из 50)

In [ ]:
m = combinations(50, 3)
m

19600

In [ ]:
p = m/ n
p

0.5727644652250146

2. Студент знает 2 вопроса и 1 не знает

In [ ]:
m1 = combinations(50, 2)
m2 = combinations(10, 1)
m_2 = m1 * m2
m_2

12250

In [ ]:
p = m_2 / n
p

0.35797779076563413

# Задача 5

Бросается игральная кость. Пусть событие A — появление чётного числа, событие B — появление числа больше трёх. Считаются ли эти события независимыми?

In [ ]:
# вероятность того, что выпало четное число больше 3 - ('4' или '6' из 6) и ('4' или '5' или '6' из 6)
pab = 4/6 * 3 / 6
pab

0.3333333333333333

In [ ]:
# вероятность того,что выпало четное число (2 4 6)
pa = 3 / 6
pa

0.5

In [ ]:
# вероятность того, что выпало число, больше 3 (4 5 6)
pb = 3 / 6
pb

0.5

In [ ]:
# если проверка выполняется, то события независимы
pab == pa * pb

False

Ответ: cобытия зависимы

# Задача 6

Допустим, есть некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.
Вопрос: какова вероятность, что вы действительно больны ей?
Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя, применив формулу полной вероятности.

In [ ]:
p_ill = 0.1 / 100
p_ill

0.001

In [ ]:
p_healthy = 1 - p_ill
p_healthy

0.999

In [ ]:
p_test_false = 1 / 100
p_test_false

0.01

In [ ]:
p_test_ok = 1 - p_test_false
p_test_ok

0.99

Вычислим полную вероятность признания больным:

In [ ]:
p_full = p_test_false * p_healthy + p_test_ok * p_ill
p_full

0.01098

Вероятность, что человек болен при результате «болен»:

In [ ]:
is_really_ill = p_test_ok * p_ill / p_full
is_really_ill

0.09016393442622951